osh/cmd_eval.py

```diff
@@ -231,8 +231,19 @@ def PlusEquals(old_val, val):
                 raise AssertionError()  # parsing should prevent this
 
         elif case(value_e.BashAssoc):
-            # TODO: Could try to match bash, it will append to ${A[0]}
-            pass
+            if tag == value_e.Str:
+                e_die("Can't append string to associative arrays")
+
+            elif tag == value_e.BashAssoc:
+                assoc_lhs = cast(value.BashAssoc, UP_old_val)
+                assoc_rhs = cast(value.BashAssoc, UP_val)
```

Comment:
Do we need to consider `tag == value_e.BashArray` ?


```
assoc+=([k]=v)  # right now we consider this BashAssoc

assoc+=(x y) # I think we consider this an array

assoc+=str
```


I guess we can add an assert for now?

Reply:
> Do we need to consider `tag == value_e.BashArray` ?

I just followed the current code for BashArray.

https://github.com/oils-for-unix/oils/blob/0df89bf3d19c70a0003db71881c45e8637199ef5/osh/cmd_eval.py#L216-L230